apps/backend/src/modules/categories/category.repository.ts

```diff
@@ -117,6 +117,37 @@ class CategoryRepository implements Repository {
 		);
 	}
 
+	public async findByIds(categoryIds: number[]): Promise<CategoryEntity[]> {
+		const categories = await this.categoryModel
+			.query()
+			.whereIn("id", categoryIds)
+			.select("*");
+
+		return await Promise.all(
+			categories.map(async (category) => {
+				const scoresModel = await this.categoryModel
+					.query()
+					.from(DatabaseTableName.QUIZ_SCORES)
+					.where({ categoryId: category.id })
+					.castTo<CategoryModel[]>();
```

Comment:
```suggestion
				const scoresModel = await this.categoryModel
					.query()
					.where({ categoryId: category.id })
					.castTo<CategoryModel[]>();
```
I believe we don't need this